### ISSUE TITLE
Feature request: nb_mark_used

### ISSUE THREAD

**Issue by Not-Nik** [TunaCici/NBBS#1]
A function for marking pages as used. This would be useful for pages that are already used by mmio, framebuffers, or the kernel itself.

**Comment by TunaCici** [comment]
Good catch! Currently there are no ways to 'reserve' a memory region to block it from being used by the allocator. It would be useful to have something like `int nb_mark_used(void *start, void *end)`. I would look into it as soon as I have time available.

Thanks for the request! And sorry I couldn't comment earlier; life got a bit crazy.
